homeassistant/components/plugwise/config_flow.py

```diff
@@ -118,9 +123,9 @@ async def async_step_zeroconf(self, discovery_info: DiscoveryInfoType):
         _name = f"{ZEROCONF_MAP.get(_product, _product)} v{_version}"
 
         self.context["title_placeholders"] = {
-            CONF_HOST: self.discovery_info[CONF_HOST],
+            CONF_HOST: self.discovery_info[zeroconf.ATTR_HOST],
             CONF_NAME: _name,
-            CONF_PORT: self.discovery_info[CONF_PORT],
+            CONF_PORT: self.discovery_info[zeroconf.ATTR_PORT],
             CONF_USERNAME: self.discovery_info[CONF_USERNAME],
         }
         return await self.async_step_user_gateway()
```

Comment:
```suggestion
        return await self.async_step_user_gateway()
        
        # PLACEHOLDER USB step_user
        async def async_step_user_gateway(self, user_input=None):
            """Handle the initial step when using network/gateway setups."""
            api = None
            errors = {}
            
            if user_input is not None:
                user_input.pop(FLOW_TYPE, None)
                
                if self.discovery_info:
                    user_input[CONF_HOST] = self.discovery_info[zeroconf.ATTR_HOST]
                    user_input[CONF_PORT] = self.discovery_info[zeroconf.ATTR_PORT]
                    user_input[CONF_USERNAME] = self._username
```

Reply:
Following your feedback, I have made some suggestions for changes.
Feel free to include these.

The last suggestion might not work, I wasn't sure how to implement this additional suggestion.

Reply:
@bouwew This suggestion adds some completely new code into this PR, that feels widely out of scope?

Reply:
I think this comment is just a bad "github ui suggestion". I have implemented the "idea" of the suggestion into the existing `async_step_user_gateway` method (see lines 146-148).

@bouwew has approved after my changes to if you are happy as well we are good to go.